db/db_iter.cc

```diff
@@ -937,7 +945,7 @@ bool DBIter::FindValueForCurrentKey() {
     // This user key has lots of entries.
     // We're going from old to new, and it's taking too long. Let's do a Seek()
     // and go from new to old. This helps when a key was overwritten many times.
-    if (num_skipped >= max_skip_ && CanReseekToSkip()) {
```

Comment:
Discussed offline. CanReseekToSkip check was already redundant here since FindValueForCurrentKeyUsingSeek is a an independent function that returns the last visible value and thus has no risk of creating loops due to difference of snapshot and max_visible_seq numbers.